eclair-core/src/main/scala/fr/acinq/eclair/db/Databases.scala

```diff
@@ -35,24 +37,39 @@ trait Databases {
 
   val pendingRelay: PendingRelayDb
 
-  def backup(file: File) : Unit
+  def backup(file: File): Unit
 }
 
-object Databases {
+object Databases extends Logging {
 
   /**
     * Given a parent folder it creates or loads all the databases from a JDBC connection
+    *
     * @param dbdir
     * @return
     */
   def sqliteJDBC(dbdir: File): Databases = {
     dbdir.mkdir()
-    val sqliteEclair = DriverManager.getConnection(s"jdbc:sqlite:${new File(dbdir, "eclair.sqlite")}")
-    val sqliteNetwork = DriverManager.getConnection(s"jdbc:sqlite:${new File(dbdir, "network.sqlite")}")
-    val sqliteAudit = DriverManager.getConnection(s"jdbc:sqlite:${new File(dbdir, "audit.sqlite")}")
-    SqliteUtils.obtainExclusiveLock(sqliteEclair) // there should only be one process writing to this file
+    var sqliteEclair: Connection = null
+    var sqliteNetwork: Connection = null
+    var sqliteAudit: Connection = null
+    try {
+      sqliteEclair = DriverManager.getConnection(s"jdbc:sqlite:${new File(dbdir, "eclair.sqlite")}")
+      sqliteNetwork = DriverManager.getConnection(s"jdbc:sqlite:${new File(dbdir, "network.sqlite")}")
+      sqliteAudit = DriverManager.getConnection(s"jdbc:sqlite:${new File(dbdir, "audit.sqlite")}")
+      SqliteUtils.obtainExclusiveLock(sqliteEclair) // there should only be one process writing to this file
+      logger.info("successful lock on eclair.sqlite")
+      databaseByConnections(sqliteAudit, sqliteNetwork, sqliteEclair)
+    } catch {
+      case t: Throwable => {
+        logger.error("could not create connection to sqlite databases: ", t)
+        if (sqliteEclair != null) sqliteEclair.close()
+        if (sqliteNetwork != null) sqliteNetwork.close()
+        if (sqliteAudit != null) sqliteAudit.close()
+        throw t
```

Comment:
Do the background jobs really need access to `eclair.sqlite`? Or only `network.sqlite` (in which case we shouldn't have any issue)?

Reply:
There are 2 background jobs, one syncing the routing table that needs access to `network.sqlite`, and one watching channels that needs access to the `eclair.sqlite` file.